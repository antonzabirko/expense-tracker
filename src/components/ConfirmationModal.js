import React from 'react';
import Modal from 'react-modal';

const ConfirmationModal = ({ modalState, onModalCancel, onModalConfirm }) => (
  <Modal
    className="modal"
    isOpen={modalState}
    onRequestClose={onModalCancel}
  >
    <div className="modal__border-container">
      <h1 className="modal__title">Remove Expense</h1>
      <p className="modal__body">Proceeding will delete and un-archive this expense.</p>
      <div className="button__container">
        <button className="button button-secondary" onClick={onModalConfirm}>Proceed</button>
        <button className="button" onClick={onModalCancel}>Cancel</button>
      </div>
    </div>
  </Modal>
);

Modal.setAppElement('#app');

export default ConfirmationModal;
