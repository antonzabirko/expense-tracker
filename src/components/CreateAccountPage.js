import React from 'react'
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'
import { withFormik, Form, Field } from 'formik'
import Recaptcha from 'react-recaptcha'
import { firebase } from '../firebase/firebase'

let firebaseError;

const CreateAccountPage = (
    {
        values,
        errors,
        touched,
        isSubmitting,
        handleChange,
    }
) => (
            <div className="box-layout">
                <div className="box-layout__color-bar"></div>
        <div className="box-layout">
            <div className="box-layout__box">
                <div className="box-layout__part">
                    <h1 className="box-layout__title">Create Account</h1>
                </div>
                <div className="box-layout__break"></div>
                <div className="box-layout__part">
                    <Form className="box-layout__form">
                        { firebaseError && <p className="box-layout__error">{firebaseError}</p> }
                        { touched.email && errors.email && <p className="box-layout__error">{errors.email}</p> }
                        <Field
                            autoComplete="username"
                            className="box-layout__input"
                            name="email"
                            placeholder="Email"
                            type="email"
                        />
                        { touched.password && errors.password && <p className="box-layout__error">{errors.password}</p> }
                        <Field
                            autoComplete="new-password"
                            className="box-layout__input"
                            name="password"
                            placeholder="Password"
                            type="password"
                        />
                        { touched.password && errors.password && <p className="box-layout__error">{errors.password}</p> }
                        <Field
                            autoComplete="new-password"
                            className="box-layout__input"
                            name="password"
                            placeholder="Confirm password"
                            type="password"
                        />
                        <div className="g-recaptcha">
                            <Recaptcha
                                className="box-layout__input g-recaptcha"
                                sitekey="6LdhzV8UAAAAAMhqlyqL-6tN5bI1SWfHdHOc9KVy"
                                theme="light"
                            />
                        </div>
                        <button className="button box-layout__button login-button box-layout__button-disabled">Sign Up!</button>
                    </Form>
                    <Link to="/" className="button button--link">Back</Link>
                </div>
            </div>
        </div>
            </div>
);

const FormikApp = withFormik({
    mapPropsToValues({ email, password}) {
        return {
            email: email || '',
            password: password || ''
        }
    },
    handleSubmit(values) {
        firebase.auth().createUserWithEmailAndPassword(values.email, values.password)
            .catch(function(error) {
                var errorCode = error.code;
                var errorMessage = error.message;
            });
    }
})(CreateAccountPage);

export default connect()(FormikApp);
