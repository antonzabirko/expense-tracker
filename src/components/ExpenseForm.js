import React from 'react';
import moment from 'moment';
import { SingleDatePicker } from 'react-dates';
import 'react-dates/lib/css/_datepicker.css';
import 'react-dates/initialize';
import uuid from 'uuid';

const now = moment();
console.log(now.format('MMM Do, YYYY'));

export default class ExpenseForm extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            description: props.expense ? props.expense.description : "",
            textArea: props.expense ? props.expense.textArea : "",
            amount: props.expense ? (props.expense.amount / 100).toString() : "",
            createdAt: props.expense ? moment(props.expense.createdAt) : moment(),
            calendarFocused: false,
            error: ''
        };
    }
    onDescriptionChange = (e) => {
        const description = e.target.value;
        this.setState(() => ({ description }));
    };
    onTextAreaChange = (e) => {
        // Alternative way:
        e.persist();
        this.setState(() => ({ textArea: e.target.value }));
    };
    onAmountChange = (e) => {
        const amount = e.target.value;
        if (!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)) {
            this.setState(() => ({ amount }));
        }
    };
    onDateChange = (createdAt) => {
        if (createdAt) {
            this.setState(() => ({ createdAt }));
        }
    };
    onFocusChange = ({ focused }) => {
        this.setState(() => ({ calendarFocused: focused }));
    };
    onSubmit = (e) => {
        e.preventDefault();
        if (!this.state.description || !this.state.amount) {
            // Error State: 'Please provide description and amount'
            this.setState(() => ({
                error: 'Please provide a description and an amount.'
            }));
        } else {
            // Clear the error
            this.setState(() => ({ error: '' }));
            this.props.onSubmit({
                description: this.state.description,
                amount: parseFloat(this.state.amount) * 100,
                createdAt: this.state.createdAt.valueOf(),
                note: this.state.textArea
            });
            console.log('submitted');
        }
    };
    render() {
        return (
            <div>
                {this.state.error &&
                    <div>
                        <p>{this.state.error}</p>
                    </div>
                }
                <form onSubmit={this.onSubmit}>
                    <input
                        type="text"
                        placeholder="Description"
                        autoFocus
                        value={this.state.description}
                        onChange={this.onDescriptionChange}
                    />
                    <input
                        type="text"
                        placeholder="Amount"
                        value={this.state.amount}
                        onChange={this.onAmountChange}
                    />
                    <SingleDatePicker
                        date={this.state.createdAt}
                        onDateChange={this.onDateChange}
                        focused={this.state.calendarFocused}
                        onFocusChange={this.onFocusChange}
                        numberOfMonths={1}
                        isOutsideRange={() => false}
                    />
                    <textarea
                        placeholder="Add a note for your expense (optional)"
                        value={this.state.textArea}
                        onChange={this.onTextAreaChange}
                    >
                    </textarea>
                    <button>Add Expense</button>
                </form>
            </div>
        )
    }
}