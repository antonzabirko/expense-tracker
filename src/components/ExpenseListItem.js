import React from 'react';

const ExpenseListItem =  ({ description, amount, createdAt }) => (
    <div>
        <p>Description: {description} | Amount: {amount} | createdAt: {createdAt}</p>
    </div>
);

export default ExpenseListItem;