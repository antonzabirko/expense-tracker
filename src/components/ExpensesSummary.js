import React from 'react';
import numeral from 'numeral';
import { connect } from 'react-redux';
import selectExpenses from "../selectors/expenses";
import expensesTotal from "../selectors/expenses-total"

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => {
    const expenseWord = expenseCount !== 1 ? 'expenses ' : 'expense';
    const expenseFormat = numeral(expensesTotal / 100).format('$0,0.00');

    return (
        <div>
            <h1>Viewing {expenseCount} {expenseWord} totalling {expenseFormat}</h1>
        </div>
    );
};

const mapStateToProps = (state) => {
    return {
        expenseCount: selectExpenses(state.expenses, state.filters).length,
        expensesTotal: expensesTotal(selectExpenses(state.expenses, state.filters))
    };
};

export default connect(mapStateToProps)(ExpensesSummary);