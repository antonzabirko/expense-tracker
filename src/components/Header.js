import React from 'react';
import { NavLink } from 'react-router-dom';

const Header = () => (
    <header>
        <h1>Expensify</h1>
        <p><NavLink to="/" exact activeClassName="is-active">Home</NavLink></p>
        <p><NavLink to="/create" activeClassName="is-active">Create</NavLink></p>
    </header>
);

export default Header;