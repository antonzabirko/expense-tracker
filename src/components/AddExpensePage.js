import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startAddExpense } from '../actions/expenses';

export class AddExpensePage extends React.Component {
  onSubmit = (expense) => {
    const { startAddExpenseConnect, history: { push } } = this.props;

    startAddExpenseConnect(expense);
    push('/');
  };

  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">
              Add Expense
            </h1>
          </div>
        </div>
        <div className="content-container">
          <ExpenseForm
            onSubmit={this.onSubmit}
          />
        </div>
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  startAddExpenseConnect: expense => dispatch(startAddExpense(expense)),
});

export default connect(undefined, mapDispatchToProps)(AddExpensePage);
