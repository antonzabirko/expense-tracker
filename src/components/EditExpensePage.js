import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startEditExpense, startRemoveExpense } from '../actions/expenses';
import ConfirmationModal from './ConfirmationModal';

export class EditExpensePage extends React.Component {
    state = {
        modalState: undefined
    };
  onSubmit = (expense) => {
      this.props.startEditExpense(this.props.expense.id, expense);
      this.props.history.push('/');
  };
  onClick = () => {
      this.setState(() => ({ modalState: true }));
  };
  onModalConfirm = () => {
      this.setState(() => ({ modalState: undefined }));
      this.props.startRemoveExpense({ id: this.props.expense.id });
      this.props.history.push('/');
  };
  onModalCancel = () => {
      this.setState(() => ({ modalState: undefined }));
  };
  render() {
    return (
      <div>
          <div className="page-header">
              <div className="content-container">
                  <h1 className="page-header__title">Edit Expense</h1>
              </div>
          </div>
          <div className="content-container">
              <ExpenseForm
                  expense={this.props.expense}
                  onSubmit={this.onSubmit}
              />
              <div>
                  <button className="button button-secondary" onClick={this.onClick}>Remove</button>
              </div>
          </div>
          <ConfirmationModal
              modalState={this.state.modalState}
              onModalConfirm={this.onModalConfirm}
              onModalCancel={this.onModalCancel}
          />
      </div>
    );
  }
}

const mapStateToProps = (state, props) => ({
        expense: state.expenses.find((expense) => (expense.id === props.match.params.id))
});

const mapDispatchToProps = (dispatch, props) => ({
  startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
  startRemoveExpense: (expense) => dispatch(startRemoveExpense({ id: expense.id }))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);
