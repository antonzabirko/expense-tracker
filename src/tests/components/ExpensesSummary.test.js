import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should render multiple expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={4} expensesTotal={1002300}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should render a single expense', () => {
    const wrapper = shallow(<ExpensesSummary Count={1} expensesTotal={40000}/>);
    expect(wrapper).toMatchSnapshot();
});