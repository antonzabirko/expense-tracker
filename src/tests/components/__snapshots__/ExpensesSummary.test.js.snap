// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a single expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container page-header-flex"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span />
       
      expenses 
       totalling 
      <span>
        $400.00
      </span>
    </h1>
    <div>
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should render multiple expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container page-header-flex"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        4
      </span>
       
      expenses 
       totalling 
      <span>
        $10,023.00
      </span>
    </h1>
    <div>
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
